### ISSUE TITLE
High weak order methods for Hamiltonian systems

### ISSUE THREAD

**Issue by ChrisRackauckas** [JuliaDiffEq/StochasticDiffEq.jl#259]
https://arxiv.org/abs/1304.3269

**Comment by jamesgardner1421** [comment]
I'd be interested in implementing the BAOAB algorithm from the reference here. Could you provide some instruction on how best to begin? I've looked at the general developer documentation but wanted to check with you first in case there are any extra specifics I should consider.

**Comment by ChrisRackauckas** [comment]
Yeah the devdocs are a good place to start to get a good feel for how it's all constructed:

http://devdocs.sciml.ai/latest/contributing/adding_algorithms/

Though this kind of system will require a partitioned system, so it might need something similar to DynamicalODEProblem for SDEs to be implemented in SciMLBase.jl. So the symplectic integrators for ODE systems + the devdocs is a good place to look, and from there, take a stab at it and I'll help you along the way.

**Comment by HadrienNU** [comment]
I will be interested to implement the remaining algorithm in the paper (ABOBA and OBABO at least).

I was thinking to implement as well the GJ set of integrator [https://arxiv.org/abs/1909.04380v1](https://arxiv.org/abs/1909.04380v1) , if this is of interest

**Comment by ChrisRackauckas** [comment]
That would be great to have, and definitely of interest.

**Comment by ChrisRackauckas** [comment]
Migrated to OrdinaryDiffEq.jl as part of StochasticDiffEq.jl archival. See https://github.com/SciML/OrdinaryDiffEq.jl/issues/3183
